core/src/error/mod.rs

```diff
@@ -70,6 +70,21 @@ impl Error {
         Error::duplicate_field(&path_to_string(path))
     }
 
+    /// Creates a new error for an attribute that is a single name-value pair, as opposed to the
+    /// expected `Meta::List` form.
+    pub fn name_value_attribute() -> Self {
+        Error::new(ErrorUnsupportedAttribute::NameValue.into())
+    }
+
+    /// Creates a new error for an attribute that `syn` failed to parse into a `Meta`, blocking
+    /// further parsing.
+    pub fn non_meta_attribute(error: &syn::Error) -> Self {
```

Comment:
Further diagnosis of the problem is out of scope for this PR, but `non_meta_attribute` is part of the public API. Instead of making this take `syn::Error`, is it better to have `darling::AttributeExt::parse_meta_list` which is implemented by `syn::Attribute`? That would allow a smaller public API for error creation, which is good, and would allow recovery logic to be baked in later without another minor version bump (also good).

Reply:
This change has been made.